test/CryptoToFiat_test.js

```diff
@@ -0,0 +1,244 @@
+    { 
+        let _conversionUpperLimit = 100
+        let depositAmount = 1200
+        var sender = accounts[5]
+        var totalClaim = 10000
+        var lastTxnBlockNumber = 0
+        await token.transfer(accounts[5], depositAmount, {from:accounts[0]});
+        await token.approve(cryptoToFiat.address, depositAmount , {from:accounts[5]}); 
+        await cryptoToFiat.deposit(depositAmount, {from:accounts[5]});
+
+        assert.equal((await cryptoToFiat.balances.call(accounts[5])).toNumber(), depositAmount)
+
+        //Create the signature
+        sender = accounts[5]
+        lastTxnBlockNumber = (await cryptoToFiat.lastTxnBlocks.call(sender)).toNumber();
+
+        let authorizer = await cryptoToFiat.authorizer.call();
+        let sgn = await signFuns.waitSignedInitiateConversionMessage(accounts[9], cryptoToFiat.address, sender, totalClaim, lastTxnBlockNumber);        
+        let vrs = signFuns.getVRSFromSignature(sgn.toString("hex"));
+
+        // Test to convert full balance - should not allow
+        lastTxnBlockNumber = (await cryptoToFiat.lastTxnBlocks.call(sender)).toNumber();
+        sgn = await signFuns.waitSignedInitiateConversionMessage(accounts[9], cryptoToFiat.address, sender, totalClaim, lastTxnBlockNumber);        
+        vrs = signFuns.getVRSFromSignature(sgn.toString("hex"));
+        testErrorRevert(cryptoToFiat.initiateConversion(depositAmount, totalClaim, vrs.v, vrs.r, vrs.s, {from:accounts[5]}));
+
+        // Test to convert more than balance amount
+        lastTxnBlockNumber = (await cryptoToFiat.lastTxnBlocks.call(sender)).toNumber();
+        sgn = await signFuns.waitSignedInitiateConversionMessage(accounts[9], cryptoToFiat.address, sender, totalClaim, lastTxnBlockNumber);        
+        vrs = signFuns.getVRSFromSignature(sgn.toString("hex"));
+        testErrorRevert(cryptoToFiat.initiateConversion(depositAmount+1, totalClaim, vrs.v, vrs.r, vrs.s, {from:accounts[5]}));
+        // Initiate the Conversion
+        var conversionAmount = 80
+        var a5_balance_b = (await cryptoToFiat.balances.call(accounts[5])).toNumber();
+        var a1_balance_b = (await cryptoToFiat.balances.call(accounts[1])).toNumber();
+        lastTxnBlockNumber = (await cryptoToFiat.lastTxnBlocks.call(sender)).toNumber();
+        sgn = await signFuns.waitSignedInitiateConversionMessage(accounts[9], cryptoToFiat.address, sender, totalClaim, lastTxnBlockNumber);        
+        vrs = signFuns.getVRSFromSignature(sgn.toString("hex"));
+        await cryptoToFiat.initiateConversion(conversionAmount, totalClaim, vrs.v, vrs.r, vrs.s, {from:accounts[5]})
+        var a5_balance_a = (await cryptoToFiat.balances.call(accounts[5])).toNumber();
+        var a1_balance_a = (await cryptoToFiat.balances.call(accounts[1])).toNumber();
+
+        assert.equal(a5_balance_a, a5_balance_b - conversionAmount)
+        assert.equal(a1_balance_a, a1_balance_b + conversionAmount)
+
+        // Test to convert immediately in the next txn
```

Comment:
I would split this test on few small tests with clear: "before" and "after" statement description. Otherwise one should read many cases from very beginning to understand why it fails or passes here.

Reply:
Okay Will divide them into multiple test cases. May be will raise one more pull request.